doc/source/tune/user-guide.rst

```diff
@@ -295,49 +295,157 @@ When running a hyperparameter search, Tune can automatically and periodically sa
+        # sync'd to head node/cloud storage
+        local_dir="/tmp/mypath",
 
-            # Obtain a checkpoint directory
-            with tune.checkpoint_dir(step=step) as checkpoint_dir:
-                path = os.path.join(checkpoint_dir, "checkpoint")
-                with open(path, "w") as f:
-                    f.write(json.dumps({"step": step}))
+        # see above! we will sync our checkpoints to S3 directory
+        sync_config=sync_config,
 
-            tune.report(hello="world", ray="tune")
+        # we'll checkpoint every 10 iterations, keep the best five
+        # checkpoints (by AUC score, descending), and always checkpoint
+        # at the end of the last run!
+        checkpoint_score_attr="max-auc",
+        keep_checkpoints_num=5,
+        checkpoint_freq=10,
+        checkpoint_at_end=True,
+
+        # a very useful trick! this will resume from the last run specified by
+        # sync_config (if one exists), otherwise it will start a new tuning run
+        resume="AUTO",
+    )
 
-    tune.run(train_func)
+In this example, checkpoints will be saved:
 
-In this example, checkpoints will be saved by training iteration to ``local_dir/exp_name/trial_name/checkpoint_<step>``.
+* **Locally**: not saved! we set ``sync_to_driver=False``, so nothing will be sync'd to the driver (your laptop)
+* **S3**: ``s3://my-checkpoints-bucket/path/my-tune-exp/<trial_name>/checkpoint_<step>``
+* **On head node**: ``~/ray-results/my-tune-exp/<trial_name>/checkpoint_<step>``
+* **On workers nodes**: ``~/ray-results/my-tune-exp/<trial_name>/checkpoint_<step>`` (but only for trials done on that node)
 
-You can restore a single trial checkpoint by using ``tune.run(restore=<checkpoint_dir>)`` By doing this, you can change whatever experiments' configuration such as the experiment's name:
+If your run stopped for any reason (finished, errored, user CTRL+C), you can restart it any time by running the script above again -- note with ``resume="AUTO"``, it will detect the previous run so long as the ``sync_config`` points to the same location.
+
+If, however, you prefer not to use ``resume="AUTO"`` (or are on an older version of Ray) you can resume manaully:
 
 .. code-block:: python
 
     # Restored previous trial from the given checkpoint
     tune.run(
-        "PG",
-        name="RestoredExp", # The name can be different.
-        stop={"training_iteration": 10}, # train 5 more iterations than previous
-        restore="~/ray_results/Original/PG_<xxx>/checkpoint_5/checkpoint-5",
-        config={"env": "CartPole-v0"},
+        # our same trainable as before
+        durable_trainer_with_params,
+
+        # The name can be different from your original name
+        name="my-tune-exp-restart",
+
+        # our same config as above!
+        restore=sync_config,
     )
 
+A simple local/rsync checkpointing example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Local or rsync checkpointing can be a good option if:
```

Comment:
Double check me on this list!

Reply:
This looks fine. I guess you can also add "if you are not interested in persistence"